src/fixate/core/checks.py

```diff
@@ -2,72 +2,187 @@
 This module is used to allow for tests to test values against criteria.
 It should implement necessary logging functions and report success or failure.
 """
-import fixate.config
+from dataclasses import dataclass
+from typing import Any, Callable, Iterable
+import logging
 
+import fixate
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass
+class CheckResult:
+    """Check Class Results to publish to subscribers
+    Is a subset of CheckClass attrs
+    """
+
+    result: bool  # Result of check
+    status: str  # Status string PASS/FAIL
+    description: str  # Description of check
+    test_val: Any = None
+    target_name: str = None  # Name of check type
+    check_string: str = None  # formatted string for UI display
+    check_params: Iterable = None  # Store for csv logging
+
+
+class _CheckClass:
```

Comment:
If it doesn't break existing code can this be changed to a dataclass?
Use fields with default values and remove the stupid init

Reply:
Done